### PR TITLE
Feat: Subscription instance increment

### PR DESCRIPTION
**Pull request by TronTram**
# Motivation / Background
- User story: https://thepressingly.atlassian.net/browse/PINET-345 
- Story technical design: https://docs.google.com/document/d/1S6e6DWKXaPXTZZsKGgyawpuoPY-p1PPv_p3XLIFcXqo/edit 
*This Pull Request has been created because:*
Reasons for the change
-  Fee from an invoice when a user consumes resources should be accurately reflected in the total value of the corresponding subscription instance

# Detail

*This Pull Request changes:*

List of things you do in this PR
- Add `IncreaseTotalValueService` to update `total_amount` of subscription instance.
- Add unit test
- Add integration test when subscription instance is created and its total amount value is incremented.
# Additional information

*TIP: Provide additional information such as screenshots, benchmarks, reference to other repositories or alternative solutions*

# Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: app/services/invoices/create_pay_in_advance_charge_service.rb [hunk 1]
```diff
@@ -144,5 +146,42 @@ def create_applied_prepaid_credit
     def refresh_amounts(credit_amount_cents:)
       invoice.total_amount_cents -= credit_amount_cents
     end
+
+    def should_increase_subscription_instance_total_value?(invoice)
+      return true if invoice.fees_amount_cents.positive?
```
Comment (tnlong311) [inline app/services/invoices/create_pay_in_advance_charge_service.rb:151]: This is more straight forward: `return invoice.fees_amount_cents.positive?`

Comment (TronTram) [reply app/services/invoices/create_pay_in_advance_charge_service.rb:151]: oh, thank you 😊

File: app/services/invoices/create_pay_in_advance_charge_service.rb [hunk 2]
```diff
@@ -144,5 +146,40 @@ def create_applied_prepaid_credit
...
+      subscription_instance_item_result.raise_if_error!
+
+      increase_total_value_result = SubscriptionInstances::IncreaseTotalValueService.call(
+        subscription_instance: current_subscription_instance,
+        fee_amount:
+      )
+
+      increase_total_value_result.raise_if_error!
+      if increase_total_value_result.success?
+        # TODO: call subscription charge service
```
Comment (ghun131) [inline app/services/invoices/create_pay_in_advance_charge_service.rb:173]: Do you mean update subscription charge?

File: app/services/invoices/create_pay_in_advance_charge_service.rb [hunk 3]
```diff
@@ -144,5 +146,40 @@ def create_applied_prepaid_credit
...
+
+    def should_increase_subscription_instance_total_value?(invoice)
+      invoice.fees_amount_cents.positive?
+    end
+
+    def increase_subscription_instance_total_value(invoice)
+      return unless current_subscription_instance
+      fee_amount = invoice.fees_amount_cents.fdiv(invoice.fees_amount.currency.subunit_to_unit)
+
+      subscription_instance_item_result = SubscriptionInstanceItems::CreateService.call(
```
Comment (ghun131) [inline app/services/invoices/create_pay_in_advance_charge_service.rb:158]: I just want to confirm that subscription instance is created and then increase total value after the creation

**Review by ghun131 — APPROVED** [review]

File: app/services/invoices/create_pay_in_advance_charge_service.rb (hunk 2 above)
Comment (TronTram) [reply app/services/invoices/create_pay_in_advance_charge_service.rb:173]: yes, that's right a

**Review by ghun131 — APPROVED** [review]

**Pull request merged**